evalml/tests/model_understanding_tests/test_partial_dependence.py

```diff
@@ -965,28 +967,37 @@ def test_partial_dependence_datetime(
 
     X = pd.DataFrame(X, columns=[str(i) for i in range(X.shape[1])])
     y = pd.Series(y)
-    random_dates = pd.Series(pd.date_range("20200101", periods=10)).sample(
-        replace=True, random_state=0, n=100
-    )
+    random_dates = pd.Series(pd.date_range("20200101", periods=size))
+    if size == 10:
+        random_dates = random_dates.sample(replace=True, random_state=0, n=100)
     random_dates.index = X.index
     X["dt_column"] = random_dates
     pipeline.fit(X, y)
-    part_dep = partial_dependence(pipeline, X, features="dt_column")
+    part_dep = partial_dependence(
+        pipeline, X, features="dt_column", grid_resolution=grid
+    )
+    expected_size = min(size, grid)
     if problem_type == "multiclass":
-        assert len(part_dep["partial_dependence"]) == 30  # 10 rows * 3 classes
-        assert len(part_dep["feature_values"]) == 30
+        assert (
+            len(part_dep["partial_dependence"]) == 3 * expected_size
+        )  # 10 rows * 3 classes
+        assert len(part_dep["feature_values"]) == 3 * expected_size
     else:
-        assert len(part_dep["partial_dependence"]) == 10
-        assert len(part_dep["feature_values"]) == 10
+        assert len(part_dep["partial_dependence"]) == expected_size
+        assert len(part_dep["feature_values"]) == expected_size
     assert not part_dep.isnull().any(axis=None)
-
-    part_dep = partial_dependence(pipeline, X, features=20)
+    # keeps the test from running too long. The part below still runs for 3 other tests
+    if grid == 100 or size == 100:
+        return
```

Comment:
This test takes about 1 min to run locally.

Reply:
@freddyaboulton The entire test takes about 1 minute locally! Leaving the grid and size unconstrained makes it run much longer